tests/step_implementers/generate_evidence/test_generate_evidence.py

```diff
@@ -0,0 +1,446 @@
+import os
+import zipfile
+from pathlib import Path
+from unittest.mock import patch
+
+import mock
+from ploigos_step_runner import StepResult
+from ploigos_step_runner.results.workflow_result import WorkflowResult
+from ploigos_step_runner.step_implementers.generate_evidence import GenerateEvidence
+from testfixtures import TempDirectory
+from tests.helpers.base_step_implementer_test_case import \
+    BaseStepImplementerTestCase
+
+
+class TestStepImplementerGenerateEvidenceBase(BaseStepImplementerTestCase):
+    def create_step_implementer(
+            self,
+            step_config={},
+            workflow_result=None,
+            parent_work_dir_path=''
+    ):
+        return self.create_given_step_implementer(
+            step_implementer=GenerateEvidence,
+            step_config=step_config,
+            step_name='generate_evidence',
+            implementer='GenerateEvidence',
+            workflow_result=workflow_result,
+            parent_work_dir_path=parent_work_dir_path
+        )
+
+class TestStepImplementerGenerateEvidence_other(TestStepImplementerGenerateEvidenceBase):
+    def test_step_implementer_config_defaults(self):
+        defaults = GenerateEvidence.step_implementer_config_defaults()
+        expected_defaults = {
+        }
+        self.assertEqual(defaults, expected_defaults)
+
+    def test__required_config_or_result_keys(self):
+        required_keys = GenerateEvidence._required_config_or_result_keys()
+        expected_required_keys = [
+            'organization',
+            'application-name',
+            'service-name',
+            'version',
+        ]
+        self.assertEqual(required_keys, expected_required_keys)
+
+class TestStepImplementerGenerateEvidence_run_step(TestStepImplementerGenerateEvidenceBase):
```

Comment:
in all of these tests you should be mocking the `__gather_evidence` function so that you are only testing the `_run_step` function and have a different set of test to independtly test the `__gather_evidence` function.

Reply:
Run_step tests are now mocking __gather_evidence and gather_evidence is being independently tested.